### PR TITLE
Add case insensitive unique sort:

### PR DESCRIPTION
**Pull request by lynxnake**
- change all sort agorithms to return 0 in case of equal strings
- fixed case insensitive sort algorithm to be really case insensitive

### CODE REVIEW

File: lib/sort-lines.js [hunk 1]
```diff
@@ -26,30 +26,30 @@ function sortLines(textEditor, startLine, endLine, algorithm, removeDuplicateVal
...
+    if (unique.length === 0 || (algorithm(unique[unique.length - 1], lines[i])) != 0) {
       unique.push(lines[i]);
     }
   }
   return unique;
 }
 
 function reverseCompare(a, b) {
-  return a < b ? 1 : -1;
+  return a = b ? 0 : a < b ? 1 : -1;
```
Comment (Tyriar) [inline lib/sort-lines.js]: These need to be `a === b`, a = b will assign

File: lib/sort-lines.js [hunk 2]
```diff
@@ -26,30 +26,30 @@ function sortLines(textEditor, startLine, endLine, algorithm, removeDuplicateVal
...
 }
 
 function caseInsensitiveCompare(a, b) {
-  return a.localeCompare(b, {sensitivity: 'base'});
+  return a.toLowerCase().localeCompare(b.toLowerCase(), {sensitivity: 'base'});
```
Comment (Tyriar) [inline lib/sort-lines.js]: Isn't `sensitivity: base` meant to do this based on the locale?

Comment (lynxnake) [reply lib/sort-lines.js]: It should do, but in my case (Win10x64, VSCode 1.20.1) it doesn't.

Comment (Tyriar) [reply lib/sort-lines.js]: @lynxnake what's your locale? Perhaps that's the reason it's not working? AFAIK toLowerCase is not how you're meant to do this from an i18n perspective.

Comment (lynxnake) [reply lib/sort-lines.js]: I have default locale:
```"locale":"en"```

Comment (lynxnake) [reply lib/sort-lines.js]: I agree that ```.toLowerCase()``` might be not the ideal solution, but in my case ```.localeCompare()``` was not working without it.

What can I do to have this pull request approved?

Comment (Tyriar) [reply lib/sort-lines.js]: @lynxnake does setting the second arg to `undefined` work? https://stackoverflow.com/a/49004987/1156119

Comment (lynxnake) [reply lib/sort-lines.js]: @Tyriar yes, it fixed this. Thanks

**Comment by Tyriar** [comment]
Thanks @lynxnake 👍

**Pull request merged**